cpp/src/arrow/io/interfaces.h

```diff
@@ -144,6 +144,15 @@ class ARROW_EXPORT RandomAccessFile : public InputStream, public Seekable {
   /// Necessary because we hold a std::unique_ptr
   ~RandomAccessFile() override;
 
+  /// \brief Create thread-safe InputStream that reads a segment of a
```

Comment:
Hmm. I think the docstring would deserve clarifying. The InputStream itself isn't thread-safe (it doesn't have a mutex inside), but you can create multiple InputStreams pointing to the same RandomAccessFile and use them concurrently (assuming each InputStream is used from a single thread).

Reply:
Right, I'll change the language